### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'introduceMe')

### ISSUE THREAD

**Issue by Abourass** [WBHarry/introduce-me#13]
After enabling the mod and trying the macro, I get the following error in DevTools:

```
VM1589:5 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'introduceMe')
    at eval (eval at _executeScript (foundry.js:17014:16), <anonymous>:5:11)
    at Macro.eval (eval at _executeScript (foundry.js:17014:16), <anonymous>:6:7)
    at Macro._executeScript (foundry.js:17016:10)
    at Macro.execute (foundry.js:16973:21)
    at MacroConfig._onExecute (foundry.js:49881:17)
```

I am using a custom game system based on the dnd5e system, but it has major changes, so if this module require the dnd5e game system then this is totally just an issue with my game system

**Comment by WBHarry** [comment]
The macros are not dependant on dnd5e. They're simply checking the module list, getting the introduce-me module and using the included API.
```
const api = game.modules.get('introduce-me').api;
await api.introduceMe(canvas.tokens.controlled[0]);
```
The error is saying that api is undefined 🤔
You mention doing major changes, but I doubt you could change something that would interfer with this. 

It feels like a silly thing to suggest, but try refreshing inside the world. The API is set up on 'initation', and I can only assume that didn't properly happen somehow. 😅

**Comment by WBHarry** [comment]
I'm closing this for now, as I can only assume the issue is due to the system complications you mentioned.

Feel free to ping it again if there's any further information.

**Comment by Abourass** [comment]
@WBHarry ->
- Updated to the newest version
- Turned on plugin
- restarted the page

Noticed that the character lacks the introduce button when right clicked so I went to the macros. These are the errors I get from them:
![image](https://user-images.githubusercontent.com/39917231/176518521-0f4700e8-bbad-40f4-be1f-ab526cd437c5.png)

**Comment by WBHarry** [comment]
Very curious. I downloaded the system, activated the mod, and everything worked fine. The button was there and the introduction went as it should.

![image](https://user-images.githubusercontent.com/89362246/177017912-f1d747bc-d043-45a9-a9d4-bfc3becc918f.png)

I think it would be interesting for you to create a new world in another system, and add the Introduce Me mod there to see if it will work for you in another system at all, or if it's a general thing for you.

Other points of interest:
- What Foundry Version are you using?
- What browser are you using?

**Comment by Abourass** [comment]
Well this is interesting! 

First, to answer your questions, I'm on the current stable version: `9.269`
I've tried going into the electron version of foundry, and tried this is in Brave (a chromium browser). 

I tried making a fresh world with my game system and just this mod:
![image](https://user-images.githubusercontent.com/39917231/178186543-679b3023-3bc1-4a60-b7a2-7c0fd57ed24d.png)

And sure enough it works! 

But then that got me thinking, what is the bug here? 

I try enabling modules, and none causes the break. 

So, I try going back to the original game world, re-enable the mod, still doesn't work. Then I try making a new character and dropping them onto the map, and sure enough they work. It seems any characters out before the mod was installed, cause the bug.
